pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -0,0 +1,266 @@
+        based on your level of memory/speed efficiency. We suggest having a look at this tutorial for
+        more information.
+        `https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html`
+
+        Arguments:
+            cpu_offload (Optional [CPUOffload]):
+                CPU offloading config. Currently, only parameter and gradient CPU
+                offload is supported. It can be enabled via passing in
+                ``cpu_offload=CPUOffload(offload_params=True)``. Note that this
+                currently implicitly enables gradient offloading to CPU in order for
+                params and grads to be on same device to work with optimizer. This
+                API is subject to change. Default is ``None`` in which case there
+                will be no offloading.
+            backward_prefetch: (Optional[BackwardPrefetch]):
+                This is an experimental feature that is subject to change in the
+                the near future. It allows users to enable two different backward_prefetch
+                algorithms to help backward communication and computation overlapping.
+                Pros and cons of each algorithm is explained in the class ``BackwardPrefetch``.
+        """
+        if not _TORCH_GREATER_EQUAL_1_11:
+            raise MisconfigurationException("DDPFullyShardedNativeStrategy is supported from pytorch v1.11.0 onwards.")
+
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            cluster_environment=cluster_environment,
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+        self._process_group = None
+        self.num_processes = len(self.parallel_devices) if self.parallel_devices is not None else 0
+        self._process_group_backend: Optional[str] = process_group_backend
+        self.cpu_offload: Optional[CPUOffload] = cpu_offload
+        self.backward_prefetch: Optional[BackwardPrefetch] = backward_prefetch
+
+    @property
+    def root_device(self) -> torch.device:
+        return self.parallel_devices[self.local_rank]
+
+    @property
+    def process_group(self) -> Optional[ProcessGroup]:
+        if self._process_group is None:
+            # The plugin should have already initilized ddp in setup_environment()
+            self._process_group = _get_default_group()
+        return self._process_group
+
+    @property
+    def setup_optimizers_in_pre_dispatch(self) -> bool:
+        # Setup optimizers after the Fully Sharded Model has been made
+        return True
+
+    @property
+    def process_group_backend(self) -> Optional[str]:
+        return self._process_group_backend
+
+    def setup_environment(self) -> None:
+        self.setup_distributed()
+        super().setup_environment()
+
+    def setup_distributed(self) -> None:
+        if not self.root_device.type == "cuda":
```

Comment:
I'm guessing here different strategies have different accelerator requirements. Maybe cleaner to let the strategy handle the requirement vs moving requirements for each strategy in the accelerator code, WDYT?

Reply:
Does native fsdp only work with GPU? 
If yes, this check should happen in acclelerator_connector to fail fast

Reply:
maybe inside the accelerator connector?

Reply:
This logic and L171 could be replaced with a check here: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/accelerator_connector.py#L564

```
if (strategy_flag == "native_fsdp" or isinstance(self._strategy_flag, DDPFullyShardedNativeStrategy)) and self._accelerator_flag == "cpu":
       raise MisconfigurationException(...)
```

Reply:
Sorry! it's https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/accelerator_connector.py#L573 here instead

Reply:
IMO it's better to keep this in the strategy. otherwise the accelerator connector becomes the union of every strategy's configuration checks, and it becomes a mess again.

Reply:
@ananthsub : can you and @four4fish sync offline and hash this out? @four4fish mentioned something related to (1) lazy init wouldn't need all env settings done during strategy init() (2) fail fast during accelerator setup (in accelerator_connector.py) as reasons to move this config check to accelerator_connector.py

Reply:
Keeping the checks in the accelerator (looks like most plugins follow the approach)